src/main/scala/io/iohk/ethereum/blockchain/data/GenesisDataLoader.scala

```diff
@@ -134,15 +118,33 @@ class GenesisDataLoader(
         Failure(new RuntimeException("Genesis data present in the database does not match genesis block from file." +
           " Use different directory for running private blockchains."))
       case None =>
-        // using empty namespace because ephemDataSource.storage already has the namespace-prefixed keys
-        ephemDataSource.storage.toSeq.grouped(dbConfig.batchSize).foreach(toStore => dataSource.update(IndexedSeq(), Nil, toStore))
+        // key.tail to remove namespace-prefix keys
```

Comment:
there should be option to remove prefix from key so GenesisDataLoader.scala does not depend on NodeStorage